src/daft-sql/src/planner.rs

```diff
@@ -756,145 +755,163 @@ impl<'a> SQLPlanner<'a> {
         Ok((exprs, desc, nulls_first))
     }
 
-    fn plan_from(&mut self, from: &[TableWithJoins]) -> SQLPlannerResult<Relation> {
+    /// Plans the FROM clause of a query and populates self.current_relation and self.table_map
+    /// Should only be called once per query.
+    fn plan_from(&mut self, from: &[TableWithJoins]) -> SQLPlannerResult<()> {
         if from.len() > 1 {
             let mut from_iter = from.iter();
 
             let first = from_iter.next().unwrap();
-            let mut rel = self.new_with_context().plan_relation(&first.relation)?;
+            let mut rel = self.plan_relation(&first.relation)?;
             self.table_map.insert(rel.get_name(), rel.clone());
             for tbl in from_iter {
-                let right = self.new_with_context().plan_relation(&tbl.relation)?;
+                let right = self.plan_relation(&tbl.relation)?;
```

Comment:
decided revert my `new_with_context` here because `SQLPlanner.plan_relation` does not mutate the plan nor does it use the alias map. Subquery relations are already handled in `plan_relation` with a `new_with_context` so this is redundant.